src/Core/Utils/SystemUtils.hpp

```
@@ -53,6 +53,20 @@ namespace SystemUtils {
 
 
+    // Concept: Has `size` method
+    template<typename T>
+    concept HasSizeMethod = requires(const T & container) {
+        { container.size() } -> std::convertible_to<std::size_t>;
+        typename T::value_type;
+    };
+
+
+    template<HasSizeMethod Container>
+    size_t ByteSize(const Container &c) {
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)